jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -453,13 +461,10 @@ public Deflater getDeflater(Request request, long content_length)
             LOG.debug("{} excluded not gzip accept {}",this,request);
             return null;
         }
-        
-        Deflater df = _deflater.get();
-        if (df==null)
-            df=new Deflater(_compressionLevel,true);        
-        else
-            _deflater.set(null);
-        
+
+        Deflater df = _deflaterPool.acquire();
+        df.setLevel(_compressionLevel);
```

Comment:
@gregw `_compressionLevel` is configurable by a setter in `GzipHandler` so should there still be an argument for the compressionLevel in the constructor

because we will still need this line incase the `_compressionLevel` changes, or maybe we could do something like `_deflaterPool.acquire(_compressionLevel);`

Reply:
@lachlan-roberts I think we should not support changes to compression level after the handler is started.   So we can just configure the deflators once and only once, not with every acquire.

Interesting with your latest jmh results, the garbage produced by the the `ConcurrentLinkedQueue` looks to be a far bit more than previously (130B/o rather than 208B/o for no pool or 64B/o for the blocking queue)? Did you change anything else?     It is possible to get a measure in your benchmark of cache hit ratio?  How often is a new deflator created vs a pooled one used and does it level off over time?

Reply:
the cache hit ratio results vary a lot depending on the size of the pool 
with 100 threads we get
```
Pool Size    Cache Hit Ratio
       10           0.437211
       20           0.657283
       50           0.952465
```